test/prompt/invocation_layer/test_sagemaker_hf_infer.py

```diff
@@ -0,0 +1,409 @@
+    via the supports() method
+    """
+    mock_client = MagicMock()
+    mock_client.describe_endpoint.return_value = {"EndpointStatus": "InService"}
+
+    mock_session = MagicMock()
+    mock_session.client.return_value = mock_client
+
+    # Patch the class method to return the mock session
+    with patch(
+        "haystack.nodes.prompt.invocation_layer.sagemaker_base.SageMakerBaseInvocationLayer.create_session",
+        return_value=mock_session,
+    ):
+        supported = SageMakerHFInferenceInvocationLayer.supports(
+            model_name_or_path="some_sagemaker_deployed_model", aws_profile_name="some_real_profile"
+        )
+    args, kwargs = mock_client.describe_endpoint.call_args
+    assert kwargs["EndpointName"] == "some_sagemaker_deployed_model"
+
+    args, kwargs = mock_session.client.call_args
+    assert args[0] == "sagemaker-runtime"
+    assert supported
+
+
+@pytest.mark.unit
+def test_supports_not_on_invalid_aws_profile_name():
+    """
+    Test that the SageMakerHFInferenceInvocationLayer raises SageMakerConfigurationError when the profile name is invalid
+    """
+
+    with patch("boto3.Session") as mock_boto3_session:
+        mock_boto3_session.side_effect = BotoCoreError()
+        with pytest.raises(SageMakerConfigurationError) as exc_info:
+            supported = SageMakerHFInferenceInvocationLayer.supports(
+                model_name_or_path="some_fake_model", aws_profile_name="some_fake_profile"
+            )
+            assert "Failed to initialize the session" in exc_info.value
+            assert not supported
+
+
+@pytest.mark.skipif(
+    not os.environ.get("TEST_SAGEMAKER_MODEL_ENDPOINT", None), reason="Skipping because SageMaker not configured"
+)
+@pytest.mark.integration
+def test_supports_triggered_for_valid_sagemaker_endpoint():
+    """
+    Test that the SageMakerHFInferenceInvocationLayer identifies a valid SageMaker Inference endpoint via the supports() method
+    """
+    model_name_or_path = os.environ.get("TEST_SAGEMAKER_MODEL_ENDPOINT")
+    assert SageMakerHFInferenceInvocationLayer.supports(model_name_or_path=model_name_or_path)
+
+
+@pytest.mark.skipif(
+    not os.environ.get("TEST_SAGEMAKER_MODEL_ENDPOINT", None), reason="Skipping because SageMaker not configured"
+)
+@pytest.mark.integration
+def test_supports_not_triggered_for_invalid_iam_profile():
+    """
+    Test that the SageMakerHFInferenceInvocationLayer identifies an invalid SageMaker Inference endpoint
+    (in this case because of an invalid IAM AWS Profile via the supports() method)
+    """
+    assert not SageMakerHFInferenceInvocationLayer.supports(model_name_or_path="fake_endpoint")
+    assert not SageMakerHFInferenceInvocationLayer.supports(
+        model_name_or_path="fake_endpoint", aws_profile_name="invalid-profile"
+    )
+
+
+@pytest.mark.unit
+def test_dolly_response_parsing(mock_auto_tokenizer, mock_boto3_session):
```

Comment:
The following tests seem very similar to me and in some cases are identical. It is up to you to decide whether to simplify/reduce them or not.

Reply:
I agree but I wanted to leave the trace that we tested all these models and their JSON response formats :-)